src/services/whatsapp.service.ts

```diff
@@ -95,66 +98,105 @@ export default class WhatsAppService {
-        {},
-        {
-          headers: this.getHeaders(),
-        },
-      ),
-    );
-    return response;
+    try {
+      const response = await firstValueFrom(
+        this.httpService.put(
+          `${this.apiUrl}/messages/${messageId}`,
+          {},
+          {
+            headers: this.getHeaders(),
+          },
+        ),
+      );
+      return response;
+    } catch (error) {
+      this.logger.warn(`ACK(seen) NOT SENT: ${error.message}`);
+    }
   }
 
-  async handleWebhook(body: any): Promise<any> {
-    const { messages } = body;
-    if (!messages) return;
-    const { id, type, from, reply } = messages[0];
-    this.sentACK(id);
-    if (type !== 'reply' || !reply?.buttons_reply?.id) return;
+  async handleWebhook(payload: any): Promise<any> {
+    if (
+      !payload ||
+      !payload.messages ||
+      !Array.isArray(payload.messages) ||
+      payload.messages.length === 0
+    ) {
+      return;
+    }
+    const { id, type, from, reply, context } = payload.messages[0];
+
+    if (!id || !from) {
+      this.logger.warn('Missing required fields in message');
+      return;
+    }
+    let appointmentId;
+    const quotedBody = context?.quoted_content?.body;
+    if (quotedBody) {
+      const idMatch = quotedBody.match(/ID:?\s*(\d+)/);
+      if (idMatch && idMatch[1]) {
+        appointmentId = idMatch[1];
+      }
+    }
+
+    if (quotedBody && !appointmentId) {
+      return;
+    }
+
+    if (type !== 'reply' || !reply?.buttons_reply?.id) {
+      return;
+    }
+
+    await this.sentACK(id);
     const numberRaw = from.slice(-10);
     const buttonId = reply.buttons_reply.id;
-    const lastAppointment =
-      await this.appointmentService.getLastAppointmentByPhone(numberRaw);
-    if (!lastAppointment) {
-      console.error('Appointment not found');
+
+    const appointment = await this.appointmentService.getById(appointmentId);
+    if (!appointment) {
+      this.logger.error(`Appointment with ID ${appointmentId} not found.`);
       return;
     }
+
     if (buttonId === 'ButtonsV3:1') {
-      await this.appointmentService.updateStatus(
-        lastAppointment.id,
-        Status.CONFIRMED,
-      );
-      const params = generateParams(
-        lastAppointment.scheduled_start,
-        lastAppointment.treatments,
-        'appointment_confirmed',
-      );
-      const formattedMessage = formatMessage(
-        messagesTemplate['appointment_confirmed'],
-        params,
-      );
-      return this.sendMessage(numberRaw, formattedMessage);
-    }
-    if (buttonId === 'ButtonsV3:2') {
-      await this.appointmentService.updateStatus(
-        lastAppointment.id,
-        Status.CANCELED,
-      );
-      const params = generateParams(
-        lastAppointment.scheduled_start,
-        lastAppointment.treatments,
-        'appointment_canceled',
+      return this.handleAppointmentStatusChange(
+        appointment,
+        numberRaw,
+        'confirmed',
```

Comment:
is this the same value as Status.CONFIRMED?, if so, I'd suggest to use the constant instead of a hardcoded value since it's easier to handle if the value of the enum changes:

```suggestion
        Status.CONFIRMED
```

Reply:
Yes, actually there is an existing enum for that

Reply:
no, that is an alias

Reply:
not an alias, you are giving a string

Reply:
```
private async handleAppointmentStatusChange(
    appointment: AppointmentResponseDto,
    phoneNumber: string,
    status: 'confirmed' | 'canceled',
  ): Promise<any> {
    const statusEnum =
      status === 'confirmed' ? Status.CONFIRMED : Status.CANCELED;
```

Reply:
an alias for your strings, I think it is better as Magallon recommend you

https://github.com/mindticas/mindticas2025-back/pull/54#discussion_r2042931033

Reply:
all right, fixed c:

Reply:
finee!!